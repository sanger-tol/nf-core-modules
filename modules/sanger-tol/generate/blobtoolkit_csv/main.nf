process GENERATE_BLOBTOOLKIT_CSV {
    tag "${meta.id}"
    label "process_low"
    executor 'local'

    input:
    tuple val(meta), path(pacbio_files)
    val(library_layout)

    output:
    tuple val(meta),    path("samplesheet.csv") , emit: csv
    path("versions.yml")                        , emit: versions

    exec:
    // Note: Manually bump version number when updating module
    def VERSION = "1.0.0"

    def fasta_files = pacbio_files.findAll { it.name.endsWith('.fasta.gz') || it.name.endsWith('.fa.gz') }

    assert fasta_files.size() > 0
    def samplesheet_entries = ["sample,datatype,datafile,library_layout"]
    fasta_files.withIndex().each{ fa, idx -> samplesheet_entries << "${meta.id}_T${idx+1},pacbio,${fa},${library_layout}" }
    file(task.workDir.resolve("samplesheet.csv")).text = samplesheet_entries.join("\n")

    file("${task.workDir}/versions.yml").text = """\
        GENERATE_BLOBTOOLKIT_CSV:
            generate_blobtoolkit_csv: ${VERSION}
        """.stripIndent()

}
