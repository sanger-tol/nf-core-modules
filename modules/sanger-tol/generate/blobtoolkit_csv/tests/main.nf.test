nextflow_process {

    name "Test Process GENERATE_BLOBTOOLKIT_CSV"
    script "../main.nf"
    process "GENERATE_BLOBTOOLKIT_CSV"

    tag "modules"
    tag "modules_sangertol"
    tag "generate"
    tag "generate/blobtoolkit_csv"

    test("fasta.gz to csv") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'resources/ascc/asccTinyTest_V2/pacbio/Pyoeliiyoelii17XNL_pbtools_simulated_pacbio_reads.fa.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'resources/treeval/TreeValTinyData/genomic_data/pacbio/seqkitPacbio50000.fasta.gz', checkIfExists: true)
                    ]
                ]
                input[1] = "SINGLE"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
