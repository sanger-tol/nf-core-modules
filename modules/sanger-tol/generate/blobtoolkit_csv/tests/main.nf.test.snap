{
    "fasta.gz to csv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "samplesheet.csv:md5,85f28ee33669df8cbe5c945988879e54"
                    ]
                ],
                "1": [
                    "versions.yml:md5,36ddfb07ce978bb13c090e8617585778"
                ],
                "csv": [
                    [
                        {
                            "id": "test"
                        },
                        "samplesheet.csv:md5,85f28ee33669df8cbe5c945988879e54"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,36ddfb07ce978bb13c090e8617585778"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-30T15:43:35.053278419"
    }
}